tests/event/events_test.php

```diff
@@ -466,4 +466,87 @@ public function test_deleting_an_element(): void {
         $this->assertEquals($templateupdatedevent->contextid, \context_system::instance()->id);
         $this->assertDebuggingNotCalled();
     }
+
+    /**
+     * Tests that the issue_created event is fired correctly.
+     */
+    public function test_issue_created_event(): void {
+        global $DB;
+
+        $this->resetAfterTest();
+
+        $course = $this->getDataGenerator()->create_course();
+        $user = $this->getDataGenerator()->create_user();
+        $customcert = $this->getDataGenerator()->create_module('customcert', ['course' => $course->id]);
+        $context = \context_module::instance($customcert->cmid);
+
+        // Simulate certificate issue.
+        $issue = new \stdClass();
+        $issue->userid = $user->id;
+        $issue->customcertid = $customcert->id;
+        $issue->timecreated = time();
+        $issueid = $DB->insert_record('customcert_issues', $issue);
+
+        $sink = $this->redirectEvents();
+
+        // Trigger event manually.
+        \mod_customcert\event\issue_created::create([
```

Comment:
Rather than triggering this manually can we please call the functions that trigger this and then check the event is called?